components/clp-package-utils/clp_package_utils/scripts/start_clp.py

```diff
@@ -1096,37 +1098,37 @@ def main(argv):
         conf_dir = clp_home / "etc"
 
         # Start components
-        if target in (ALL_TARGET_NAME, DB_COMPONENT_NAME):
+        if target in (ALL_TARGET_NAME, DB_COMPONENT_NAME) and DB_COMPONENT_NAME not in exclude:
```

Comment:
I personally don't like how the code is written now. it's quite hard to read and could be error prone. In addition, it looks like if user make a typo in the exclude list, CLP will sliently run the "exclude target"

In the closed source CLP, we have a config file to specify what components to run and one can remove component from the config file to remove components.
It might make more sense if we let the start-clp to launch everything by default (i.e. if a config file is not provided), or launch the specified component based on the config file.

If we plan to go with the current approach, I think if the if statement can be viewed as

`if target in (ALL_TARGET, [TASK_TARGET]) and TASK_TARGET not in exclude:`

maybe we can factor it out as a function to make the code less messy.

Reply:
> I personally don't like how the code is written now. it's quite hard to read and could be error prone. In addition, it looks like if user make a typo in the exclude list, CLP will sliently run the "exclude target"
> 
> In the closed source CLP, we have a config file to specify what components to run and one can remove component from the config file to remove components. It might make more sense if we let the start-clp to launch everything by default (i.e. if a config file is not provided), or launch the specified component based on the config file.
> 
> If we plan to go with the current approach, I think if the if statement can be viewed as
> 
> `if target in (ALL_TARGET, [TASK_TARGET]) and TASK_TARGET not in exclude:`
> 
> maybe we can factor it out as a function to make the code less messy.

With help from @Henry8192 , we have refactored the if-checks into calls of a checker helper. Also now we also validate whether the names in the `exclude` list are valid before proceeding. Let me know if this is more readable and maintainable now.